loader/icd_dispatch.c

```diff
@@ -99,6 +99,12 @@ clGetDeviceIDs(cl_platform_id   platform,
 {
     // initialize the platforms (in case they have not been already)
     khrIcdInitialize();
+
+    if (!platform && khrIcdVendors != NULL)
```

Comment:
These changes for a `NULL` platform are fine according to the spec, but this will mean that different versions of the ICD loader will have different behavior for a `NULL` platform.  I'd be more comfortable discussing them as part of a separate PR.

Reply:
Removing for now to unblock rest of the changes. Let's discuss this separately.